### PR TITLE
fix: Refactor `extension.ts` 

### PR DESCRIPTION
**Pull request by zFernand0**
## Proposed changes

Split the `extension.ts` file into multiple `init.ts` files in their corresponding folders (dataset, shared, ...)

## Release Notes

Milestone: 2.7.0

Changelog: N/A

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [x] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

File: packages/zowe-explorer/src/utils/LoggerUtils.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import * as globals from "../globals";
+
+// Set up localization
+nls.config({
+    messageFormat: nls.MessageFormat.bundle,
+    bundleFormat: nls.BundleFormat.standalone,
+})();
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+
+export async function initializeZoweLogger(context: vscode.ExtensionContext): Promise<void> {
```
Comment (traeok) [inline packages/zowe-explorer/src/utils/LoggerUtils.ts:24]: Same scenario as `ProfileUtils.ts` - not asking you to make any changes, just thinking out loud for future codebase refactoring 😅

File: packages/zowe-explorer/src/uss/init.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+import * as ussActions from "./actions";
+import * as refreshActions from "../shared/refresh";
+import { IZoweUSSTreeNode, IZoweTreeNode, IZoweTree } from "@zowe/zowe-explorer-api";
+import { Profiles } from "../Profiles";
+import * as contextuals from "../shared/context";
+import { getSelectedNodeList } from "../shared/utils";
+import { createUSSTree } from "./USSTree";
+import { initSubscribers } from "../shared/init";
+
+export async function initUSSProvider(context: vscode.ExtensionContext) {
```
Comment (traeok) [inline packages/zowe-explorer/src/uss/init.ts:23]: Do we want to make this a default export since its the only exported function in this file?

**Review by traeok — APPROVED** [review]
LGTM, thanks @zFernand0 - just had a couple questions that I wanted to ask regarding this

**Review by rudyflores — APPROVED** [review]
LGTM! Thanks for this refactor @zFernand0 !

File: packages/zowe-explorer/src/utils/LoggerUtils.ts (hunk 1 above)
Comment (zFernand0) [reply packages/zowe-explorer/src/utils/LoggerUtils.ts:24]: I definitely like the idea of a singleton or class-with-static methods 😋

File: packages/zowe-explorer/src/uss/init.ts (hunk 2 above)
Comment (zFernand0) [reply packages/zowe-explorer/src/uss/init.ts:23]: no strong preference.
I guess I do like [this comment on stackoverflow](https://stackoverflow.com/a/33307487/16555983) about code-refactoring made easier if using named exports ( no `default` keyword) 😋

File: packages/zowe-explorer/src/utils/ProfilesUtils.ts [hunk 3]
```diff
@@ -347,3 +348,20 @@ export function writeOverridesFile() {
         fs.writeFileSync(settingsFile, fileContent, "utf8");
     }
 }
+
+export async function initializeZoweProfiles(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (zFernand0) [reply packages/zowe-explorer/src/utils/ProfilesUtils.ts:352]: Definitely for a future PR 😋

**Comment by zFernand0** [comment]
Thank you @traeok and @rudyflores for the reviews
And thanks @JillieBeanSim for merging right away 🙏 

You guys don't know how much work this saves by merging now 😋 
I'm happy to help address any merge conflicts that anyone may have as a result of this PR being merged 😉 
Just let me know

**Pull request merged**